src/parser.cpp

```diff
@@ -891,24 +892,24 @@ void Parser::ParserImpl::loadConnection(const ModelPtr &model, const XmlNodePtr
     ComponentPtr component1 = nullptr;
     ComponentPtr component2 = nullptr;
     // Now check the objects exist in the model. TODO Remove as is validation?
-    if (model->containsComponent(componentNamePair.first)) {
-        component1 = model->component(componentNamePair.first);
+    if (model->containsComponent(componentNamePair.at(0))) {
```

Comment:
I would replace `.at(0)` with `[0]`. (Same elsewhere.)

Indeed, `.at()` is only really needed if you provide an index value that may be out of rance (in which it will raise an exception). Here, you know that the index value is fine, so no need for the "extra work" that `.at()` does.

Reply:
052f1676